scripts/preinstall.js

```diff
@@ -17,7 +19,32 @@ const configText = `module.exports = {
   localPlugins: [],
 }`;
 
+// simple utility to remove whitespace from string
+function trim(string) {
+  return string.replace(/^\s+|\s+$/g, '');
+}
+
 try {
+  // check minimum version of npm
+  let npmVersion = execSync('npm --version', {
```

Comment:
This could fail if the user invokes a version of `npm` that isn't the first on the `PATH`

Reply:
Any suggestions? Presumably if they are doing that then we still want to fail since the plugin build process will also use the same npm that this is invoking anyway and so we probably still want this to fail.

Reply:
I've been digging for something to suggest, but its turning into nasty bash hackery.
`npm` should be in the users path so I don't think that it will be a problem for most, I'm thinking that it could cause a problem inside of docker potentially in some cases.
Maybe we leave a comment here and allow for the edge case to be handled programmatically by using an env var `BPANEL_NPM_PATH` if it exists and if it doesn't, then just using regular old `npm`? I think that this flexibility is useful so that developers won't have to change the code if having `npm` hardcoded doesn't work for them. What do you think?

Reply:
I like giving flexibility to developers so that is why I suggested the env var thing, but I'm not particularly strongly opinionated on it. I was trying to namespace the env var so that it wouldn't collide. Hopefully this edge case doesn't happen. We can add the `PATH` gotcha to the docs